### PR TITLE
windows10-icons: init at 1.0

### LINKED ISSUE #237726: Package request: B00merang-Artwork/Windows-10
**Project description**

The Windows 10 Icon theme repository

**Metadata**

* homepage URL: http://b00merang.weebly.com/windows-10.html
* source URL: https://github.com/B00merang-Artwork/Windows-10
* license: gpl3
* platforms: unix, linux, darwin

### PR DESCRIPTION
**Pull request by mibmo**
###### Description of changes

Add Windows 10 icon theme from B00merang artwork - https://github.com/B00merang-Artwork/Windows-10

Closes #237726

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/data/icons/windows10-icons/default.nix
Comment (apfelkuchen6) [inline pkgs/data/icons/windows10-icons/default.nix:21]:
```suggestion
    license = licenses.unfree;
```
These icons have been ripped out of Windows, they are certainly not licensed under the GPL.

Comment (mibmo) [reply pkgs/data/icons/windows10-icons/default.nix:21]: Very fair, I was originally thinking the same (see comments on issue this closes). Are we still allowed to include it?

Comment (apfelkuchen6) [reply pkgs/data/icons/windows10-icons/default.nix:21]: I'm not really sure.

There is lots of unfree software in nixpkgs. Marking software as unfree results in the CI not building it and hence it not being available in the NixOS binary cache. It will be transparently built on the users' machines.

For this reason I think including anything in nixpkgs should be legally fine as long as it's marked as unfree -- NixOS infrastructure won't touch it, so it essentially is the same as just linking to the project. I'm not a lawyer and this is not legal advice.

Whether we want to include things that are pretty much blatant copyright infringement in nixpkgs is another thing. I can't find a precedent for this at first glance, but there probably has been one.

Comment (mibmo) [reply pkgs/data/icons/windows10-icons/default.nix]: Good point. I'm not really sure I want to be the one opening the flood gates and setting precedent either, but nonetheless I've committed the change - we'll see whether this gets merged or not I suppose. :)

File: pkgs/data/icons/windows10-icons/default.nix
Comment (emilylange) [inline pkgs/data/icons/windows10-icons/default.nix:22]: Please add your user to https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix first.

See https://nixos.org/manual/nixpkgs/unstable/#var-meta-maintainers

**Review by emilylange — CHANGES_REQUESTED** [review]

File: pkgs/data/icons/windows10-icons/default.nix
Comment (mibmo) [reply pkgs/data/icons/windows10-icons/default.nix:22]: My first PR #237651  - where I do add myself to the maintainers list - is going through the review process right now. Is it best to wait till it gets merged and FF this branch or should I add it to this PR too? (my concern is mainly merge conflicts, but if it's no problemo then I'll just copy the commit :))

Comment (emilylange) [reply pkgs/data/icons/windows10-icons/default.nix:22]: Whoops, sorry. Totally missed the comment where you already mentioned that^^

Since you asked:
You could base this PR on top of https://github.com/NixOS/nixpkgs/pull/237651/commits/189e0ded75b46a9ab4bed91cb213744eae92e6ad from the other PR.

But it shouldn't matter all that much. The easiest would be to just wait for the other PR to get merged first :)

File: pkgs/data/icons/windows10-icons/default.nix
Comment (emilylange) [inline pkgs/data/icons/windows10-icons/default.nix:9]: Please use `fetchFromGitHub` instead :)

**Review by emilylange — CHANGES_REQUESTED** [review]

File: pkgs/data/icons/windows10-icons/default.nix
Comment (mibmo) [reply pkgs/data/icons/windows10-icons/default.nix:9]: I've no idea how that slipped me! :) Thank you, I'll make the change.

File: pkgs/data/icons/windows10-icons/default.nix
Comment (Freed-Wu) [reply pkgs/data/icons/windows10-icons/default.nix:24]: #237651 has been merged. So I believe this change has been solved. :smile:

Comment (mibmo) [reply pkgs/data/icons/windows10-icons/default.nix:24]: Totally right! I'll resolve and get the PR ready ^^

File: pkgs/by-name/wi/windows10-icons/package.nix
Comment (emilylange) [inline pkgs/by-name/wi/windows10-icons/package.nix:12]:
```suggestion

  dontConfigure = true;
  dontBuild = true;

```

File: pkgs/by-name/wi/windows10-icons/package.nix
Comment (emilylange) [inline pkgs/by-name/wi/windows10-icons/package.nix:18]: Have you tested your derivation?
It looks like its output does not contain even a single icon:

```
❯ tree result
result
└── share
    └── icons
        └── windows10
            ├── 128x128
            ├── 16x16
            ├── 22x22
            ├── 24x24
            ├── 256x256
            ├── 32x32
            ├── 48x48
            ├── 512x512
            ├── 8x8
            ├── index.theme
            ├── README.md
            └── scalable

14 directories, 2 files
```

**Review by emilylange — CHANGES_REQUESTED** [review]

**Comment by emilylange** [comment]
Hi @mibmo, are you still interested in working on this? :eyes: 

I marked your PR as draft for now.

**Comment by mibmo** [comment]
> Hi @mibmo, are you still interested in working on this? 👀
> 
> I marked your PR as draft for now.

Yep, sorry! Been down with _the virus_ the past few weeks, barely been able to do anything. I'll pick it up once I have time again, thanks for checking in :)

**Comment by Freed-Wu** [comment]
I add it to NUR before this PR is accepted :smile:

**Review by emilylange — APPROVED** [review]

**Comment by Freed-Wu** [comment]
I advice to keep same name as other distribution. In archlinux's AUR, the folder name is `Windows10`. in nixpkgs, it is `windows10`. I am not sure which should be standard. https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=windows10-icon-theme#n22 https://github.com/NixOS/nixpkgs/pull/237972

**Pull request merged**
